### DISCUSSION TITLE
Set persistent password - not working for me

### DISCUSSION THREAD

**Discussion by DMSnz** [rustdesk/rustdesk#5454] [General]
As per the guides found here;
https://rustdesk.com/docs/en/self-host/pro/faq/#can-i-deploy-rustdesk-using-powershell

How can I set a persistent password ....

I have tried this against client builds 1.2.0 and 1.2.1 without success.
Testing with Rustdesk installed to one of my servers
Run the powershell script interactively with admin rights and all looks fine, script runs and it writes out the 12 character password without issue.
But upon attempting to connect to this host the password is not accepted.

I am not sure why this does not work, anyone else tested this successfully ??

**Comment by dinger1986** [comment]
That script doesn’t set a password? I have successfully done it with my rmm but that’s using system not admin. I have been testing with 1.2.2

**Comment by DMSnz** [comment]
Maybe that is the difference, I am running it with domain admin logged in, powershell run in admin mode but no joy.
I am looking at TRMM as we type ;)

**Reply by dinger1986** [reply]
Cool. Works well, I’ll do testing tomorrow and revert :)

**Reply by DMSnz** [reply]
I will test 1.2.2 now to be sure also ... post back soon

**Reply by DMSnz** [reply]
same issue, 1.2.2 refuses to accept that generated password

**Reply by DMSnz** [reply]
I will push it out as a PS script via GPO, computer side not user side, so runs as system.
But I will try modify the script to give it a pre set password, not randomly generated one, else will need to record the output somewhere and I am guessing you have captured that info with TRMM?

**Reply by DMSnz** [reply]
sorry for spam amount of posts but here is a solution for my preference of having a self created password
I replaced the password generation line with simply;
$rustdesk_pw = 'YOURPASSWORD'

To test on my server I created a scheduled task to run this powershell script, set to run as system, as I could not reboot my server right now to test GPO method, but proven now so happy days

**Reply by dinger1986** [reply]
all makes sense, as the service is ran as SYSTEM so to write the password is should also be as SYSTEM same as `--get-id`

Yes I use TRMM to collect it for me
